aim/sdk/adapters/optuna.py

```diff
@@ -0,0 +1,192 @@
+        should only be used with ``study.optimize(n_jobs=1)``.
+
+
+    Args:
+        metric_name:
+            Name assigned to optimized metric. In case of multi-objective optimization,
+            list of names can be passed. Those names will be assigned
+            to metrics in the order returned by objective function.
+            If single name is provided, or this argument is left to default value,
+            it will be broadcasted to each objective with a number suffix in order
+            returned by objective function e.g. two objectives and default metric name
+            will be logged as ``value_0`` and ``value_1``. The number of metrics must be
+            the same as the number of values objective function returns.
+        as_multirun:
+            Creates new runs for each trial. Useful for exploring parameters in Aim UI
+            (for more: https://aimstack.readthedocs.io/en/latest/ui/pages/explorers.html#params-explorer).
+
+    """
+
+    def __init__(
+        self,
+        metric_name: Union[str, Sequence[str]] = "value",
+        as_multirun: bool = False,
+        repo: Optional[str] = None,
+        experiment_name: Optional[str] = None,
+        system_tracking_interval: Optional[int] = DEFAULT_SYSTEM_TRACKING_INT,
+        log_system_params: bool = True,
+    ) -> None:
+
+        _imports.check()
+
+        if not isinstance(metric_name, Sequence):
+            raise TypeError(
+                "Expected metric_name to be string or sequence of strings, got {}.".format(
+                    type(metric_name)
+                )
+            )
+
+        self._metric_name = metric_name
+        self._as_multirun = as_multirun
+        self._repo_path = repo
+        self._experiment_name = experiment_name
+        self._system_tracking_interval = system_tracking_interval
+        self._log_system_params = log_system_params
+        self._run = None
+        self._run_hash = None
+
+        if not self._as_multirun:
+            self.setup()
+
+    def __call__(self, study: optuna.study.Study, trial: optuna.trial.FrozenTrial) -> None:
+
+        if isinstance(self._metric_name, str):
+            if len(trial.values) > 1:
+                # Broadcast default name for multi-objective optimization.
+                names = ["{}_{}".format(self._metric_name, i) for i in range(len(trial.values))]
+
+            else:
+                names = [self._metric_name]
+
+        else:
+            if len(self._metric_name) != len(trial.values):
+                raise ValueError(
+                    "Running multi-objective optimization "
+                    "with {} objective values, but {} names specified. "
+                    "Match objective values and names, or use default broadcasting.".format(
+                        len(trial.values), len(self._metric_name)
+                    )
+                )
+
+            else:
+                names = [*self._metric_name]
```

Comment:
should `names = self._metric_name` be enough?

Reply:
As the `metric_name` is `Union[str, Sequence[str]]` we write a generic operation for all types of sequences.

Reply:
I mean just Line 96, not the whole logic of getting names.
as an Iterable `[*self._metric_name]` and `self._metric_name` are identical no?

Reply:
Sure, sorry haven't notices the usage of the `names` variable on the further lines.